From c12906559bebd1dc9dde3a46741c3742c58255c8 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Tue, 2 Oct 2018 15:20:04 +0900
Subject: [PATCH 14/28] nsis: Fix some !ifdef conditions

Newly add HAVE_VIS_VIM and make it disabled by default.
VisVim is for VC 6.0.  Does anyone still use it?
---
 nsis/gvim.nsi | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index 9be7d3fb0..dace234b9 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -36,6 +36,9 @@ Unicode true
 # Comment the next line if you do not want to add Native Language Support
 !define HAVE_NLS
 
+# Uncomment the next line if you want to include VisVim extension:
+#!define HAVE_VIS_VIM
+
 # Comment the following line to create a multilanguage installer:
 !define HAVE_MULTI_LANG
 
@@ -44,7 +47,9 @@ Unicode true
 # ----------- No configurable settings below this line -----------
 
 !include "Library.nsh"		# For DLL install
-!include "UpgradeDLL.nsh"	# for VisVim.dll
+!ifdef HAVE_VIS_VIM
+  !include "UpgradeDLL.nsh"	# for VisVim.dll
+!endif
 !include "LogicLib.nsh"
 !include "MUI2.nsh"
 !include "nsDialogs.nsh"
@@ -587,6 +592,7 @@ SectionGroup $(str_group_plugin) id_group_plugin
 SectionGroupEnd
 
 ##########################################################
+!ifdef HAVE_VIS_VIM
 Section "$(str_section_vis_vim)" id_section_visvim
 	SectionIn 3
 
@@ -594,6 +600,7 @@ Section "$(str_section_vis_vim)" id_section_visvim
 	!insertmacro UpgradeDLL "${VIMSRC}\VisVim\VisVim.dll" "$0\VisVim.dll" "$0"
 	File ${VIMSRC}\VisVim\README_VisVim.txt
 SectionEnd
+!endif
 
 ##########################################################
 !ifdef HAVE_NLS
@@ -673,10 +680,12 @@ Section -post
 	  SectionGetSize ${id_section_editwith} $4
 	  IntOp $3 $3 + $4
 	${EndIf}
+!ifdef HAVE_VIS_VIM
 	${If} ${SectionIsSelected} ${id_section_visvim}
 	  SectionGetSize ${id_section_visvim} $4
 	  IntOp $3 $3 + $4
 	${EndIf}
+!endif
 !ifdef HAVE_NLS
 	${If} ${SectionIsSelected} ${id_section_nls}
 	  SectionGetSize ${id_section_nls} $4
@@ -836,10 +845,12 @@ Section "un.$(str_unsection_register)" id_unsection_register
 	# created.  Thus the "vim61" directory is included in it.
 	StrCpy $0 "$INSTDIR"
 
+!ifdef HAVE_VIS_VIM
 	# If VisVim was installed, unregister the DLL.
 	${If} ${FileExists} "$0\VisVim.dll"
 	  ExecWait "regsvr32.exe /u /s $0\VisVim.dll"
 	${EndIf}
+!endif
 
 	# delete the context menu entry and batch files
 	nsExec::Exec "$0\uninstal.exe -nsis"
@@ -910,7 +921,9 @@ Section "un.$(str_unsection_exe)" id_unsection_exe
 	RMDir /r $0\syntax
 	RMDir /r $0\tools
 	RMDir /r $0\tutor
+!ifdef HAVE_VIS_VIM
 	RMDir /r $0\VisVim
+!endif
 	RMDir /r $0\lang
 	RMDir /r $0\keymap
 	Delete $0\*.exe
-- 
2.17.0

